### PR TITLE
fix(@angular-devkit/core): resolving global npm modules was not working is some situations

### PR DESCRIPTION
**Pull request by ymeine**
Examples:

- when setting custom npm prefix
- when using nodist under windows (it is actually also setting custom npm prefix)

Therefore, use `npm config get prefix` to get the actual prefix, keep the environment variable `PREFIX` priority and fallback to other methods if not working.

### CODE REVIEW

**Review by hansl — DISMISSED** [review]

**Comment by hansl** [comment]
Please fix the test error (seems like there's a last case missing?) and this can be merged.

File: packages/angular_devkit/core/node/resolve.ts [hunk 1]
```diff
@@ -50,17 +51,22 @@ function _getGlobalNodeModules() {
...
-      globalPrefix = path.join(destdir, globalPrefix);
+    const npmPrefix = _getNpmPrefix();
+    if (npmPrefix != null) {
+      globalPrefix = npmPrefix;
+    } else if (process.platform === 'win32') {
+      // c:\node\node.exe --> prefix=c:\node\
+      globalPrefix = path.dirname(process.execPath);
+    } else {
+      // /usr/local/bin/node --> prefix=/usr/local
+      globalPrefix = path.dirname(path.dirname(process.execPath));
```
Comment (hansl) [inline packages/angular_devkit/core/node/resolve.ts]: I think this might be why it's failing on circle. Their execPath might be different.

**Comment by ymeine** [comment]
Hello, 

I know I come back quite late, unfortunately I don't have time to work on this task lately. I did a specific commit to log some information and here it is: 

- `which node` ⇒ `/usr/local/bin/node`
- `node -pe "process.execPath"` ⇒ `/usr/local/bin/node`
- `which npm` ⇒ `/usr/local/bin/npm`
- `npm config get prefix` ⇒ `/home/circleci/.npm-global`
- `node -pe "try {require.resolve('npm');} catch (e) {''}"` ⇒ nothing
- `ls /usr/local/lib/node_modules` ⇒ gives a single entry `npm`
- `ls /home/circleci/.npm-global/node_modules` ⇒ triggers an access error since the folder doesn't exist

How to interpret it: 

- the `node` binary is in folder `/usr/local/bin` and so is the `npm` one
- the global modules path as configured through npm is `/home/circleci/.npm-global` but it seems to have no module installed in it (since there is no `node_modules` folder inside)
- resolving `npm` from Node.js code doesn't give anything, which is consistent with the fact there is apparently no module installed globally
- however we can see there there is actually a `npm` folder in `/usr/local/lib/node_modules` and that's why the test used to pass since that's an implemented use case (but the code I added takes precedence and makes it fail)

Leaving this comment here for now, hoping one could tackle this in the future.

**Comment by alexeagle** [comment]
Sorry @ymeine  - this got lost when we moved back to the angular/angular-cli repo. If you still think this pull request is relevant, could you please re-base on that repo and open a new PR there? Thanks and sorry for the extra churn.

**Pull request closed without merging**